### PR TITLE
feat(cb2-5699): create TRL and HGV weight sections

### PR DESCRIPTION
**Pull request by naathanbrown**
## Amend HGV & TRL Weights section for a provisional Tech Record

Create the sections for the HGV and TRL sections and make them ammendable.
[https://dvsa.atlassian.net/browse/CB2-5699](5699)

## Checklist

- [x] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [x] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [x] Squashed commits contain the JIRA ticket number
- [x] Link to the PR added to the repo
- [x] Delete branch after merge

### CODE REVIEW

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/forms/components/weights-trl-hgv/weights-trl-hgv.component.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  templateUrl: './weights-trl-hgv.component.html'
+})
+export class WeightsTrlHgvComponent extends WeightsComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() vehicleType!: string;
+
+  constructor(public override dfs: DynamicFormService) {
+    super(dfs);
+  }
+
+  override ngOnInit() {
```
Comment (gjulien-bjss) [inline src/app/forms/components/weights-trl-hgv/weights-trl-hgv.component.ts:16]: Do you need to override these? (genuinely don't know just looks weird)

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts [hunk 2]
```diff
@@ -74,17 +74,14 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
...
+    errors.length ? this.errorService.setErrors(errors) : this.errorService.clearErrors();
 
     return forms.some(form => form.invalid);
   }
 
   handleFormState() {
-    const form = this.summary.sections
-    .map(section => section.form)
-    .concat(this.summary.dimensions.form, this.summary.weights.form);
+    const weightForm = this.summary.weightsTrlHgv?.form ? this.summary.weightsTrlHgv.form : this.summary.weights.form;
```
Comment (tom-bjss) [inline src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:83]: Im thinking this check should probably be against the vehicle type on the evaluation, rather then checking if the form exists 😄

File: src/app/forms/components/weights-trl-hgv/weights-trl-hgv.component.ts (hunk 1 above)
Comment (naathanbrown) [reply src/app/forms/components/weights-trl-hgv/weights-trl-hgv.component.ts:16]: Yeah I agree it looks weird, appears to be a TS ruling, I can turn it off in the TS config but for the sake of a key word I'm okay with it.

File: src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts (hunk 2 above)
Comment (naathanbrown) [reply src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts:83]: I can do this, but the current tech record is an observable at this point, so I would have to subscribe every single time this method is called, which feels like adding massive overhead for very little gain. Unless you see a better way of doing it?

**Review by kpearmanbjss — APPROVED** [review]

**Review by kpearmanbjss — DISMISSED** [review]

**Review by gjulien-bjss — DISMISSED** [review]

File: src/app/forms/services/dynamic-form.service.ts [hunk 3]
```diff
@@ -124,14 +124,17 @@ export class DynamicFormService {
...
+      meta: { name, label, customValidatorErrorName }
     } = control;
 
     if (errors) {
       const errorList = Object.keys(errors);
 
       errorList.forEach(error => {
-        validationErrorList.push({ error: ErrorMessageMap[error](errors[error], label), anchorLink: name } as GlobalError);
+        validationErrorList.push({
+          error: ErrorMessageMap[error](errors[error], customValidatorErrorName ? customValidatorErrorName : label),
```
[the remark this answers is not in the document]
Comment (naathanbrown) [reply src/app/forms/services/dynamic-form.service.ts:135]: I knew there was syntax for this but couldn't remember it lmao, thanks

**Review by tahayazici — DISMISSED** [review]

**Review by tom-bjss — APPROVED** [review]

**Review by tom-bjss — DISMISSED** [review]

**Review by aaron-downing — APPROVED** [review]

**Review by tahayazici — APPROVED** [review]

**Pull request merged**